devito/types/dense.py

```diff
@@ -1454,3 +1453,160 @@ def parent(self):
+        if pointer_dim is not None:
+            halo = ((0, 0),) + halo
+        return halo
+
+    @property
+    def data(self):
+        # Any attempt at allocating data by the user should fail miserably
+        raise TypeError("TempFunction cannot allocate data")
+
+    data_domain = data
+    data_with_halo = data
+    data_ro_domain = data
+    data_ro_with_halo = data
+
+    @property
+    def pointer_dim(self):
+        return self._pointer_dim
+
+    @property
+    def dim(self):
+        return self.pointer_dim
+
+    @property
+    def shape(self):
+        domain = [i.symbolic_size for i in self.dimensions]
+        return DimensionTuple(*domain, getters=self.dimensions)
+
+    @property
+    def shape_with_halo(self):
+        domain = self.shape
+        halo = [sympy.Add(*i, evaluate=False) for i in self._size_halo]
+        ret = tuple(sum(i) for i in zip(domain, halo))
+        return DimensionTuple(*ret, getters=self.dimensions)
+
+    shpe_allocated = DiscreteFunction.symbolic_shape
+
+    def make(self, shape=None, initializer=None, allocator=None, **kwargs):
+        """
+        Create a Function which can be used to override this TempFunction
+        in a call to `op.apply(...)`.
+
+        Parameters
+        ----------
+        shape : tuple of ints, optional
+            Shape of the domain region in grid points.
+        initializer : callable or any object exposing the buffer interface, optional
+            Data initializer. If a callable is provided, data is allocated lazily.
+        allocator : MemoryAllocator, optional
+            Controller for memory allocation. To be used, for example, when one wants
+            to take advantage of the memory hierarchy in a NUMA architecture. Refer to
+            `default_allocator.__doc__` for more information.
+        **kwargs
+            Mapper of Operator overrides. Used to automatically derive the shape
+            if not explicitly provided.
+        """
+        if shape is None:
+            if len(kwargs) == 0:
+                raise ValueError("Either `shape` or `kwargs` (Operator overrides) "
+                                 "must be provided.")
+            shape = []
+            for n, i in enumerate(self.shape):
+                v = i.subs(kwargs)
+                if not v.is_Integer:
+                    raise ValueError("Couldn't resolve `shape[%d]=%s` with the given "
+                                     "kwargs (obtained: `%s`)" % (n, i, v))
+                shape.append(int(v))
+            shape = tuple(shape)
+        elif len(shape) != self.ndim:
+            raise ValueError("`shape` must contain %d integers, not %d"
+                             % (self.ndim, len(shape)))
+        elif not all(is_integer(i) for i in shape):
+            raise ValueError("`shape` must contain integers (got `%s`)" % str(shape))
+
+        return Function(name=self.name, dtype=self.dtype, dimensions=self.dimensions,
```

Comment:
So someone can get a standard `Function` back, and therefore try to take FD along the pointer dimension?

Reply:
it's a Function so you can do whatever with it and its dimensions, just like any other Function with custom Dimensions

Note that none of this is ever created unless one specifies `Operator(... opt=..., {'cire-ftemps': True})` . Ie these TempFunctions are a user-requested feature, so if a user ever gets to this point, she/he should be well aware of how to use the Functions created by TempFunction's `make`